todo-list/script.js

```diff
@@ -1,150 +1,231 @@
-        // wrapperList.removeEventListener('click', handlerCheckedTodo);
-        const id = parseInt(event.target.closest('li').getAttribute('id'), 10);
-        const todoText = app.querySelector(`li[id='${id}'] .todo-text`);
-        const todoEdit = app.querySelector(`li[id='${id}'] .todo-edit`);
-
-        // hidden true todo text
-        todoText.hidden = true;
-        // hidden false input edite
-        todoEdit.hidden = false;
-
-        // replace this value input to todo list array
-        function changeValueOfTodo(event) {
-            todos[id].label = this.value;
-            renderTodos();
-
-            // Clean the memory
-            todoEdit.hidden = true;
-            todoEdit.removeEventListener('change', changeValueOfTodo);
-        }
-        todoEdit.addEventListener('change', changeValueOfTodo);
+    // helper method
+    createElement(elementName, value='', addClass='', id='') {
+        const element =  document.createElement(elementName);
+        if(addClass) element.classList.add(addClass);
+        if(value) element.innerText = value;
+        if(id) element.setAttribute('id', id + '');
+        return element;
+    }
 
-        function handelerDoesntChaneValue(event) {
-            if(event.key === 'Enter' && todos[id].label == this.value) {
-                todoEdit.hidden = true;
-                renderTodos();
+    createStructureApp() {
+        // create wrapper app
+        const wrapperElementApp = this.createElement('section','','wrapper-todo-app');
+
+        // create Title
+        const h1 = this.createElement('h1');
+        h1.innerText = this.titleOfApp;
+        wrapperElementApp.appendChild(h1);
+
+        const wrapperInsertTask = this.createElement('section','', 'wrapper-insert');
+        const addTaskBtn = this.createElement('button', 'Add Task', 'add-task-btn');
+        const inputElement = this.createElement('input', '', 'input-text-task');
+        inputElement.setAttribute('type','text');
+        inputElement.setAttribute('placeholder','Enter New Task');
+        wrapperInsertTask.appendChild(addTaskBtn);
+        wrapperInsertTask.appendChild(inputElement);
+        wrapperElementApp.appendChild(wrapperInsertTask);
+
+        const wrapperListOfTasks = this.createElement('ul', '', 'wrapper-list-taks');
+        wrapperElementApp.appendChild(wrapperListOfTasks);
+
+        // get context this for use inside functions
+        const mainContext = this;
+        // function of event
+        function handlerAddTaskBtn(event) {
+            let value = '';
+            // console.log(this.tagName)
+            if(this.tagName == 'INPUT') {
+                value = this.value.trim();
+            } else if(this.tagName == 'BUTTON') {
+                value = this.closest('section').querySelector('.input-text-task').value;         
+            }
+            
+            const result = Boolean(mainContext.state.find(taskObj => taskObj.label == value));
```

Comment:
You may also use [Array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)